src/examples/learn_quant/conf/config.py

```diff
@@ -0,0 +1,29 @@
+import dataclasses
+from dataclasses import dataclass, field
+from omegaconf import DictConfig
+import hydra
+from hydra.core.config_store import ConfigStore
+
+@dataclasses.dataclass
+class ModeConfig:
+    name: str
+
+@dataclasses.dataclass
+class UniverseConfig:
```

Comment:
OK, so having read everything now, the point is: `depth` is used by `generate_unique_expressions`, which is part of the grammar, to generate expressions.  IT's not a property of the `Universe`, so that argument should be moved elsehwere in the Config, i.e. to a `GrammarConfig` sub-part or some such